i-slide.js

```diff
@@ -843,7 +843,22 @@ class ISlide extends HTMLElement {
   }
 }
 
-// Register the custom element
-customElements.define('i-slide', ISlide);
-
 export default ISlide;
+
+// We enable side-effects when hard-coded query string parameters are set
+if (import.meta && import.meta.url) {
+  const queryString = new URL(import.meta.url).search.slice(1);
+  const params = new URLSearchParams(queryString);
+  if (params.get("register") !== null || params.get("replaceSelector")) {
+    customElements.define('i-slide', ISlide);
```

Comment:
The need to add a `register` param to have the script do the one thing I'd like it to do, meaning handle `<i-slide>` elements, does not necessarily strike me as a fantastic default. I get the willingness not to introduce any side effect, detailed for instance in the [Supporting both automatic and manual registration of custom elements](https://component.kitchen/blog/posts/supporting-both-automatic-and-manual-registration-of-custom-elements) blog post. Is that common practice though? Could that rather be done the other way round, meaning "register by default unless `manualRegistration` is set"?

Reply:
This was triggered by a recent discussion on Twitter https://twitter.com/LeaVerou/status/1463500111275769857 which shows that at least it's a somewhat common concern.

I could live with the reverse approach, but I'm not convinced it's better; in particular, it relies on a side-effect (the import URL) to disable side-effects.

Reply:
But what will Twitter say about it next week? ;)

Why do you call a URL parameter a side-effect?

I just view that as two distinct versions of the code, available at two different URLs, which I think also matches what Jake Archibald mentions in this thread. Actually, I would prefer the [suggested convention from Nick Williams](https://twitter.com/WickyNilliams/status/1463602443988705288): something like `ISlide.js` for the class itself and `i-slide.js` for the version with side effects. The possible collision mentioned in the thread can arise no matter what as far as I can tell, there does not seem to be much that can be done about it.

The `selector` parameter would be applicable to `i-slide.js` in that convention, which would make it clearer that the code needs to auto-register itself through `customElements.define` in that case.

Reply:
ok - so are you suggesting I create a new `ISlide.js` with just the class, and `i-slide.js` which imports the class? or should `i-slide.js` also incorporate the code of the class (through some to be defined tooling)?

Reply:
(the issue with the former is that it requires another HTTP request; not sure how costly that is in practice with HTTP2+)

Reply:
(the URL param is a side-effect from my perspective because it depends on code execution anchored in how the script was loaded via `import.meta.url`, which wouldn't be the case for an actual different URL)

Reply:
Using an import seems good to me.

I realize the whole thing is slightly more painful due to the need to update the logic in [`publish-lib.js`](https://github.com/w3c/i-slide/blob/main/tools/publish-lib.js) to also create versioned flavors of `iSlide.js`.

Reply:
I've done the split (but not updated publish-lib.js yet) - if you can already review that part…

Reply:
and a9abfd7 should deal with the publishing script